server/opts.go

```diff
@@ -2685,6 +2733,103 @@ func parseTLS(v interface{}) (t *TLSConfigOpts, retErr error) {
 	return &tc, nil
 }
 
+func parseWebsocket(v interface{}, o *Options, errors *[]error, warnings *[]error) error {
+	var lt token
+	defer convertPanicToErrorList(&lt, errors)
+
+	tk, v := unwrapValue(v, &lt)
+	gm, ok := v.(map[string]interface{})
+	if !ok {
+		return &configErr{tk, fmt.Sprintf("Expected websocket to be a map, got %T", v)}
+	}
+	for mk, mv := range gm {
+		// Again, unwrap token value if line check is required.
+		tk, mv = unwrapValue(mv, &lt)
+		switch strings.ToLower(mk) {
+		case "listen":
+			hp, err := parseListen(mv)
+			if err != nil {
+				err := &configErr{tk, err.Error()}
+				*errors = append(*errors, err)
+				continue
+			}
+			o.Websocket.Host = hp.host
+			o.Websocket.Port = hp.port
+		case "port":
+			o.Websocket.Port = int(mv.(int64))
+		case "host", "net":
+			o.Websocket.Host = mv.(string)
+		case "tls":
+			config, _, err := getTLSConfig(tk)
+			if err != nil {
+				*errors = append(*errors, err)
+				continue
+			}
+			o.Websocket.TLSConfig = config
+		case "same_origin":
+			o.Websocket.SameOrigin = mv.(bool)
+		case "allowed_origins":
+			switch mv := mv.(type) {
+			case string:
+				o.Websocket.AllowedOrigins = []string{mv}
+			case []interface{}:
+				keys := make([]string, 0, len(mv))
+				for _, val := range mv {
+					tk, val = unwrapValue(val, &lt)
+					if key, ok := val.(string); ok {
+						keys = append(keys, key)
+					} else {
+						err := &configErr{tk, fmt.Sprintf("error parsing allowed origins: unsupported type in array %T", val)}
+						*errors = append(*errors, err)
+						continue
+					}
+				}
+				o.Websocket.AllowedOrigins = keys
+			default:
+				err := &configErr{tk, fmt.Sprintf("error parsing allowed origins: unsupported type %T", mv)}
+				*errors = append(*errors, err)
+			}
+			// Disable compression for now...
```

Comment:
Would remove for now instead of commenting out.

Reply:
Just here or are you suggesting to remove everything related to compression (options and code doing it)? It is a lot to remove and will definitively have to save a branch so I don't have to re-do this work again..

Reply:
Done